graviti/operation/frame.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Definitions of different operations on a DataFrame."""
+
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+from graviti.openapi import add_data, update_data
+
+if TYPE_CHECKING:
+    from graviti.dataframe import DataFrame
+
+_MAX_BATCH_SIZE = 2000
+
+
+class DataFrameOperation:
+    """This class defines the basic method of the operation on a DataFrame."""
+
+    def do(  # pylint: disable=invalid-name
+        self,
+        access_key: str,
+        url: str,
+        owner: str,
+        dataset: str,
+        *,
+        draft_number: int,
+        sheet: str,
+    ) -> None:
+        """Execute the OpenAPI create sheet.
+
+        Arguments:
+            access_key: User's access key.
+            url: The URL of the graviti website.
+            owner: The owner of the dataset.
+            dataset: Name of the dataset, unique for a user.
+            draft_number: The draft number.
+            sheet: The sheet name.
+
+        Raises:
+            NotImplementedError: The method of the base class should not be called.
+
+        """
+        raise NotImplementedError
+
+
+class AddData(DataFrameOperation):
+    """This class defines the operation that add data to a DataFrame.
+
+    Arguments:
+        data: The data to be added.
+        strategy_arguments: Arguments required by the ``__record_key`` generation strategy
+
+    """
+
+    def __init__(
+        self, data: "DataFrame", strategy_arguments: Optional[Dict[str, Any]] = None
+    ) -> None:
+        self.data = data
+        self.strategy_arguments = strategy_arguments
+
+    def do(  # pylint: disable=invalid-name
+        self,
+        access_key: str,
+        url: str,
+        owner: str,
+        dataset: str,
+        *,
+        draft_number: int,
+        sheet: str,
+    ) -> None:
+        """Execute the OpenAPI add data.
+
+        Arguments:
+            access_key: User's access key.
+            url: The URL of the graviti website.
+            owner: The owner of the dataset.
+            dataset: Name of the dataset, unique for a user.
+            draft_number: The draft number.
+            sheet: The sheet name.
+
+        """
+        data = self.data.to_pylist()
+        for position in range(0, len(data), _MAX_BATCH_SIZE):
```

Comment:
we have a method named `chunked` in `tensorbay/utility/itertools.py`, can we reuse it?